.circleci/config.yml

```diff
@@ -91,13 +91,13 @@ env-linux-medium: &env-linux-medium
   NUMBER_OF_NINJA_PROCESSES: 3
 
 env-linux-2xlarge: &env-linux-2xlarge
-  NUMBER_OF_NINJA_PROCESSES: 18
+  NUMBER_OF_NINJA_PROCESSES: 34
```

Comment:
1. Where does the `2 * cores + 2` heuristic come from? Not disagreeing with it; just asking to learn

2. It seems kind of brittle to code the current value instead of the heuristic that was used to calculate it. Would it be possible to use something like this instead:

```sh
NUMBER_OF_CORES=`cat /proc/cpuinfo | grep processor | wc -l`
NUMBER_OF_NINJA_PROCESSES=$(expr $NUMBER_OF_CORES + $NUMBER_OF_CORES + 2)
```

IIUC this would work as a one-stop setting regardless off the size profile

Reply:
@ckerr The `2 * cores + 2` comes from ninja's built in default.  The reason we have to hard code this and not calculate on the fly is because the processor count in CircleCI is reported incorrectly which causes ninja to think it can spin up like 80 processes and they eventually deadlock.

So in answer to `2` we unfortunately have to hardcode these numbers.

Reply:
The answer to `2` appears to be that we have to provide some numbers, but it's not clear from what you said that we have to hardcode them.

Do either `grep -c ^processor /proc/cpuinfo` or `nproc` return a usable number in these environments?

Reply:
@ckerr They return incorrect numbers, that's what made us hard code these in the first place.  Something about circleci's platform makes those numbers incorrect.